server/webapp/WEB-INF/rails.new/spec/controllers/api_v1/admin/environments_controller_spec.rb

```diff
@@ -101,6 +121,42 @@
         expect(controller).to disallow_action(:get, :show, name: @environment_name).with(401, 'You are not authorized to perform this action.')
       end
     end
+
+    describe :route do
+      describe :with_header do
+        before :each do
+          Rack::MockRequest::DEFAULT_ENV["HTTP_ACCEPT"] = "application/vnd.go.cd.v1+json"
+        end
+        after :each do
+          Rack::MockRequest::DEFAULT_ENV = {}
+        end
+        it 'should route to show action of environments controller for alphanumeric environment name' do
```

Comment:
The only drawback I see with this approach is that the testing scenario  is not explicit,

eg: ```it('should route to ... for pipeline name with dot')``` somebody will have to see the test data to understand the scenarios being tested.